Defining `VULKAN_HPP_NO_EXCEPTIONS` when using C++ modules results in missing `vk::raii` members

I'm experimenting with the C++ module and I have created the following working setup (the example one in the README.md doesn't work):

```cmake
# RToolBox\cmake\VulkanHPP.cmake
include_guard()

# find Vulkan SDK
find_package("Vulkan" REQUIRED)

# set up Vulkan C++ module
add_library("VulkanCppModule")
add_library("vulkan::cppm" ALIAS "VulkanCppModule")

target_compile_definitions("VulkanCppModule"
  PUBLIC
    # Disable exceptions
    "VULKAN_HPP_NO_EXCEPTIONS"
    # Enable designated initializers
    "VULKAN_HPP_NO_STRUCT_CONSTRUCTORS"

)
target_compile_features("VulkanCppModule"
  PUBLIC
    cxx_std_23
)
target_include_directories("VulkanCppModule"
  PRIVATE
    "${Vulkan_INCLUDE_DIR}"
)
target_link_libraries("VulkanCppModule"
  PUBLIC
    "Vulkan::Vulkan"
)

target_sources("VulkanCppModule"
  PUBLIC
    FILE_SET cxx_modules TYPE CXX_MODULES
    BASE_DIRS
      "${Vulkan_INCLUDE_DIR}"
    FILES
      "${Vulkan_INCLUDE_DIR}/vulkan/vulkan.cppm"
)
```

Build errors:

```text
[build] vulkan.cppm
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2915): error C2039: 'Context': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2915): error C2873: 'Context': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2916): error C2039: 'ContextDispatcher': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2916): error C2873: 'ContextDispatcher': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2917): error C2039: 'DeviceDispatcher': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2917): error C2873: 'DeviceDispatcher': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2918): error C2039: 'exchange': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2918): error C2873: 'exchange': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2919): error C2039: 'InstanceDispatcher': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2919): error C2873: 'InstanceDispatcher': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2926): error C2039: 'Buffer': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2926): error C2873: 'Buffer': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2927): error C2039: 'BufferView': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2927): error C2873: 'BufferView': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2928): error C2039: 'CommandBuffer': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2928): error C2873: 'CommandBuffer': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2929): error C2039: 'CommandBuffers': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2929): error C2873: 'CommandBuffers': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2930): error C2039: 'CommandPool': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2930): error C2873: 'CommandPool': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2931): error C2039: 'DescriptorPool': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2931): error C2873: 'DescriptorPool': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2932): error C2039: 'DescriptorSet': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2932): error C2873: 'DescriptorSet': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2933): error C2039: 'DescriptorSetLayout': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2933): error C2873: 'DescriptorSetLayout': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2934): error C2039: 'DescriptorSets': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2934): error C2873: 'DescriptorSets': symbol cannot be used in a using-declaration
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2935): error C2039: 'Device': is not a member of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2909): note: see declaration of 'vk::raii'
[build] C:\SDKs\VulkanSDK\1.3.261.0\Include\vulkan\vulkan.cppm(2935): error C2873: 'Device': symbol cannot be used in a using-declaration

... etc...
```

--- a/VulkanHppGenerator.cpp
+++ b/VulkanHppGenerator.cpp
@@ -725,9 +725,11 @@ export namespace VULKAN_HPP_NAMESPACE
 {
   ${usings}
 
+#if !defined( VULKAN_HPP_DISABLE_ENHANCED_MODE ) && !defined( VULKAN_HPP_NO_EXCEPTIONS )
   namespace VULKAN_HPP_RAII_NAMESPACE {
     ${raiiUsings}
   } // namespace VULKAN_HPP_RAII_NAMESPACE
+#endif
 } // namespace VULKAN_HPP_NAMESPACE
 )";
 

--- a/vulkan/vulkan.cppm
+++ b/vulkan/vulkan.cppm
@@ -2929,6 +2929,7 @@ export namespace VULKAN_HPP_NAMESPACE
   using VULKAN_HPP_NAMESPACE::isObsoletedExtension;
   using VULKAN_HPP_NAMESPACE::isPromotedExtension;
 
+#if !defined( VULKAN_HPP_DISABLE_ENHANCED_MODE ) && !defined( VULKAN_HPP_NO_EXCEPTIONS )
   namespace VULKAN_HPP_RAII_NAMESPACE
   {
     //======================
@@ -3027,10 +3028,10 @@ export namespace VULKAN_HPP_NAMESPACE
     //=== VK_NV_device_generated_commands ===
     using VULKAN_HPP_RAII_NAMESPACE::IndirectCommandsLayoutNV;
 
-#if defined( VK_USE_PLATFORM_FUCHSIA )
+#  if defined( VK_USE_PLATFORM_FUCHSIA )
     //=== VK_FUCHSIA_buffer_collection ===
     using VULKAN_HPP_RAII_NAMESPACE::BufferCollectionFUCHSIA;
-#endif /*VK_USE_PLATFORM_FUCHSIA*/
+#  endif /*VK_USE_PLATFORM_FUCHSIA*/
 
     //=== VK_EXT_opacity_micromap ===
     using VULKAN_HPP_RAII_NAMESPACE::MicromapEXT;
@@ -3043,4 +3044,5 @@ export namespace VULKAN_HPP_NAMESPACE
     using VULKAN_HPP_RAII_NAMESPACE::ShaderEXTs;
 
   }  // namespace VULKAN_HPP_RAII_NAMESPACE
+#endif
 }  // namespace VULKAN_HPP_NAMESPACE

--- a/vulkan/vulkansc.cppm
+++ b/vulkan/vulkansc.cppm
@@ -1403,6 +1403,7 @@ export namespace VULKAN_HPP_NAMESPACE
   using VULKAN_HPP_NAMESPACE::isObsoletedExtension;
   using VULKAN_HPP_NAMESPACE::isPromotedExtension;
 
+#if !defined( VULKAN_HPP_DISABLE_ENHANCED_MODE ) && !defined( VULKAN_HPP_NO_EXCEPTIONS )
   namespace VULKAN_HPP_RAII_NAMESPACE
   {
     //======================
@@ -1471,10 +1472,11 @@ export namespace VULKAN_HPP_NAMESPACE
     //=== VK_EXT_debug_utils ===
     using VULKAN_HPP_RAII_NAMESPACE::DebugUtilsMessengerEXT;
 
-#if defined( VK_USE_PLATFORM_SCI )
+#  if defined( VK_USE_PLATFORM_SCI )
     //=== VK_NV_external_sci_sync2 ===
     using VULKAN_HPP_RAII_NAMESPACE::SemaphoreSciSyncPoolNV;
-#endif /*VK_USE_PLATFORM_SCI*/
+#  endif /*VK_USE_PLATFORM_SCI*/
 
   }  // namespace VULKAN_HPP_RAII_NAMESPACE
+#endif
 }  // namespace VULKAN_HPP_NAMESPACE

